backend/server/db/helpers/sessions.py

```diff
@@ -0,0 +1,100 @@
+        return GuestSessionInfoModel(
+            uid=session["uid"],
+            curr_ref_token=RefreshToken(session["currRefreshToken"]),
+            expires_at=exp,
+            type="guest",
+        )
+
+    assert session["type"] == "csesoc"
+    return SessionInfoModel(
+        uid=session["uid"],
+        curr_ref_token=RefreshToken(session["currRefreshToken"]),
+        oidc_info=SessionOIDCInfoModel(
+            access_token=session["oidcInfo"]["accessToken"],
+            raw_id_token=session["oidcInfo"]["rawIdToken"],
+            refresh_token=session["oidcInfo"]["refreshToken"],
+            validated_id_token=session["oidcInfo"]["validatedIdToken"],
+        ),
+        expires_at=exp,
+        type="csesoc",
+    )
+
+def insert_not_setup_session(sid: SessionID, info: NotSetupSessionModel) -> bool:
+    # NOTE: should only be used to reserve the sid,
+    # needed since we need a unique session id allocated before setting refresh token info
+    # and we need a refresh token to set a full session info
+    try:
+        sessionsNewCOL.insert_one({
+            "sid": sid,
+            "uid": info.uid,
+            "expiresAt": datetime.datetime.fromtimestamp(info.expires_at, tz=datetime.timezone.utc),
+            "type": "notsetup",
+        })
+        return True
+    except pymongo.errors.DuplicateKeyError:
+        # sid already existed
+        return False
+
+def update_csesoc_session(sid: SessionID, expires_at: PositiveInt, curr_ref_token: RefreshToken, info: SessionOIDCInfoModel) -> bool:
+    res = sessionsNewCOL.update_one(
+        { "sid": sid, "type": { "$in": [ "csesoc", "notsetup" ] } },
+        {
+            "$set": {
+                "expiresAt": datetime.datetime.fromtimestamp(expires_at, tz=datetime.timezone.utc), 
+                "currRefreshToken": curr_ref_token,
+                "type": "csesoc",
+                "oidcInfo": {
+                    "accessToken": info.access_token,
+                    "rawIdToken": info.raw_id_token,
+                    "refreshToken": info.refresh_token,
+                    "validatedIdToken": info.validated_id_token,
+                },
+            },
+        },
+        upsert=False,
+        hint="sidIndex",
+    )
+
+    return res.matched_count == 1
+
+def update_guest_session(sid: SessionID, expires_at: PositiveInt, curr_ref_token: RefreshToken) -> bool:
+    res = sessionsNewCOL.update_one(
+        { "sid": sid, "type": { "$in": [ "guest", "notsetup" ] } },
+        {
+            "$set": {
+                "expiresAt": datetime.datetime.fromtimestamp(expires_at, tz=datetime.timezone.utc), 
+                "currRefreshToken": curr_ref_token,
+                "type": "guest",
+            },
+        },
+        upsert=False,
+        hint="sidIndex",
+    )
+
+    return res.matched_count == 1
+
+def delete_all_sessions(sid: SessionID) -> bool:
+    res = sessionsNewCOL.delete_many({ "sid": sid })
+    return res.deleted_count > 0
```

Comment:
Makes more sense for this function to return the count. Callers that were going to do check the return value can do their own checks for `> 0` - or even keep the assert / condition the same.

Reply:
this is actually interesting, as there should only be one session per sid (as is enforced by a unique sid index)... I think I had the delete_many for sanity.

Should I convert to delete_one and change it to return `res.deleted_count == 1`?

Reply:
If we are sure that there will be only one - which I think we are able to enforce using the database itself right?
In that case, `delete_one` should be fine / preferred

Reply:
agreed now